### PR TITLE
Add total wait time to thread pool in nodes stats

### LINKED ISSUE #5007: [DOC] New threadpool metric thread_pool.total_wait_time
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 A new metric `thread_pool.total_wait_time` is being introduced for 2.11

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
PR: 
- https://github.com/opensearch-project/OpenSearch/pull/9681

Issue:
- https://github.com/opensearch-project/OpenSearch/issues/9645

### PR DESCRIPTION
**Pull request by kolchfa-aws**
Fixes #5007 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _api-reference/nodes-apis/nodes-stats.md [hunk 1]
```diff
@@ -977,6 +979,7 @@ active | Integer | The number of active threads in the pool.
 rejected | Integer | The number of tasks that have been rejected.
 largest | Integer | The peak number of threads in the pool.
 completed | Integer | The number of tasks completed.
+total_wait_time | Integer | The total time tasks spent waiting in the thread pool queue.
```
Comment (jed326) [inline _api-reference/nodes-apis/nodes-stats.md:982]: Not all `thread_pool`s support this metric today -- only `search`, `search_throttled`, and `index_searcher` thread_pools support it. Not sure what the right place to mention this (if at all) is

**Review by jed326 — APPROVED** [review]

File: _api-reference/nodes-apis/nodes-stats.md (hunk 1 above)
Comment (kolchfa-aws) [inline _api-reference/nodes-apis/nodes-stats.md:982]:
```suggestion
total_wait_time | Integer | The total time tasks spent waiting in the thread pool queue. Currently, only `search`, `search_throttled`, and `index_searcher` thread pools support this parameter.
```

Comment (jed326) [reply _api-reference/nodes-apis/nodes-stats.md]: `parameter` -> `metric`

File: _api-reference/nodes-apis/nodes-stats.md [hunk 2]
```diff
@@ -977,6 +979,7 @@ active | Integer | The number of active threads in the pool.
 rejected | Integer | The number of tasks that have been rejected.
 largest | Integer | The peak number of threads in the pool.
 completed | Integer | The number of tasks completed.
+total_wait_time | Integer | The total time tasks spent waiting in the thread pool queue. Currently, only `search`, `search_throttled`, and `index_searcher` thread pools support this parameter.
```
Comment (kolchfa-aws) [inline _api-reference/nodes-apis/nodes-stats.md:982]:
```suggestion
total_wait_time | Integer | The total time tasks spent waiting in the thread pool queue. Currently, only `search`, `search_throttled`, and `index_searcher` thread pools support this metric.
```

File: _api-reference/nodes-apis/nodes-stats.md [hunk 3]
```diff
@@ -27,8 +27,8 @@ The following table lists the available path parameters. All path parameters are
 Parameter | Type | Description
 :--- | :--- | :---
 nodeId | String | A comma-separated list of nodeIds used to filter results. Supports [node filters]({{site.url}}{{site.baseurl}}/api-reference/nodes-apis/index/#node-filters). Defaults to `_all`.
-metric | String | A comma-separated list of metric groups that will be included in the response. For example, `jvm,fs`. See the list of all metrics below. Defaults to all metrics.
-index_metric | String | A comma-separated list of index metric groups that will be included in the response. For example, `docs,store`. See the list of all index metrics below. Defaults to all index metrics.
+metric | String | A comma-separated list of metric groups that will be included in the response. For example, `jvm,fs`. See the following list of all index metrics. Defaults to all metrics.
```
Comment (hdhalter) [inline _api-reference/nodes-apis/nodes-stats.md:30]:
```suggestion
metric | String | A comma-separated list of metric groups that are included in the response. For example, `jvm,fs`. See the following list of all index metrics. Defaults to all metrics.
```

File: _api-reference/nodes-apis/nodes-stats.md [hunk 4]
```diff
@@ -27,8 +27,8 @@ The following table lists the available path parameters. All path parameters are
...
+index_metric | String | A comma-separated list of index metric groups that will be included in the response. For example, `docs,store`. See the following list of all index metrics. Defaults to all index metrics.
```
Comment (hdhalter) [inline _api-reference/nodes-apis/nodes-stats.md:31]:
```suggestion
index_metric | String | A comma-separated list of index metric groups that are included in the response. For example, `docs,store`. See the following list of all index metrics. Defaults to all index metrics.
```

File: _api-reference/nodes-apis/nodes-stats.md [hunk 5]
```diff
@@ -977,6 +979,7 @@ active | Integer | The number of active threads in the pool.
 rejected | Integer | The number of tasks that have been rejected.
 largest | Integer | The peak number of threads in the pool.
 completed | Integer | The number of tasks completed.
+total_wait_time | Integer | The total time tasks spent waiting in the thread pool queue. Currently, only `search`, `search_throttled`, and `index_searcher` thread pools support this metric.
```
Comment (natebower) [inline _api-reference/nodes-apis/nodes-stats.md:982]:
```suggestion
total_wait_time | Integer | The total amount of time tasks spent waiting in the thread pool queue. Currently, only `search`, `search_throttled`, and `index_searcher` thread pools support this metric.
```

File: _api-reference/nodes-apis/nodes-stats.md [hunk 6]
```diff
@@ -1066,7 +1069,7 @@ total.count | Integer | The total number of documents ingested by the node.
 total.time_in_millis | Integer | The total amount of time for preprocessing ingest documents, in milliseconds.
 total.current | Integer | The total number of documents that are currently being ingested by the node.
 total.failed | Integer | The total number of failed ingestions for the node.
-pipelines | Object | Ingest pipeline statistics for the node. Each pipeline is a nested object specified by its ID with the properties listed below.
+pipelines | Object | Ingest pipeline statistics for the node. Each pipeline is a nested object specified by its ID with the following properties.
```
Comment (natebower) [inline _api-reference/nodes-apis/nodes-stats.md:1072]: Should "with" be "and has", as on line 1108?

**Review by natebower — COMMENTED** [review]
@kolchfa-aws One change and one comment. Thanks!

File: _api-reference/nodes-apis/nodes-stats.md [hunk 7]
```diff
@@ -977,6 +979,7 @@ active | Integer | The number of active threads in the pool.
 rejected | Integer | The number of tasks that have been rejected.
 largest | Integer | The peak number of threads in the pool.
 completed | Integer | The number of tasks completed.
+total_wait_time | Integer | The total amount of time tasks spent waiting in the thread pool queue. Currently, only `search`, `search_throttled`, and `index_searcher` thread pools support this metric.
```
Comment (sohami) [inline _api-reference/nodes-apis/nodes-stats.md:982]: @kolchfa-aws @jed326 We are saying that this is supported for `search` threadpool but in the example above it is showing up for `ad-batch-task-threadpool` and `OPENSEARCH_ML_TASK_THREAD_POOL`. So either we add search threadpool stats in above example with this new metric or remove the new metric from those threadpool

**Pull request merged**